### PR TITLE
go 1.19 migration

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/terraform/structure/terraform_parser_test.go [hunk 1]
```diff
@@ -215,7 +214,7 @@ func TestTerrraformParser_Module(t *testing.T) {
 					}
 				}
 				if !isYorTagExists {
-					t.Error(fmt.Sprintf("tag not found on merged block %v", yorTagKey))
+					t.Errorf("tag not found on merged block %v", yorTagKey)
```
Comment (nimrodkor) [inline src/terraform/structure/terraform_parser_test.go:217]: @JamesWoolfenden 😱

**Review by nimrodkor — COMMENTED** [review]
Very nice @JamesWoolfenden !
What's up with all the whitespace changes? 😅 

Also, can we take a shot at replacing stuff from src/common/utils/utils.go with built in functions, if any?

File: src/terraform/structure/terraform_parser_test.go (hunk 1 above)
Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: ive added some new pre-commit hooks - gosimple was complaining about sprintf, fair enough?

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: the white space changes are pre-commit enforcing a consistent approach to spacing and line termination.

Comment (nimrodkor) [reply src/terraform/structure/terraform_parser_test.go:217]: This is AMAZINGGGGG
Wasn't available in 1.17 I believe. this is a shock of amazement

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: errorf seems so obvious i thought it was a dev mistake, didnt know it was new

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: we could replace /utils but personally if it aint broke?

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: we should probably update and fix all the GHA actions? Some refer to latest which we think is bad these days.

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: e.g. actions/checkout@v2 where current is https://github.com/actions/checkout is v3.3

Comment (JamesWoolfenden) [reply src/terraform/structure/terraform_parser_test.go:217]: and where i fixed the versions in the dockerfiles to FROM alpine:3.17.1

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**